aten/src/ATen/native/Distributions.cpp

```diff
@@ -487,6 +487,17 @@ Tensor& multinomial_out(Tensor& result, const Tensor& self, int64_t n_sample, bo
   } else {
     result.resize_({n_sample});
   }
+
+  // Fast-path based on RobertoLat example.
+  // Reference:
+  // https://github.com/pytorch/pytorch/issues/11931#issuecomment-625882503
+  // Half is not supported on CPU.
+  if (!with_replacement &&
+      !(self.device().is_cpu() && self.scalar_type() == ScalarType::Half)) {
+    auto rand = at::rand_like(self);
+    std::tie(std::ignore, result) = (rand.log() / self).topk(n_sample);
```

Comment:
This is actually not going to work, you have to in-place copy to result, not reassign reference. 
Also, @pbelevich, in the line above we don't pass `gen` argument to `rand_like` function, which is also probably wrong - can you please take a look? What would be the correct way to generate a tensor filled with uniformly distributed number using provided `gen`?

Reply:
Since `at::rand_like(self)` literally does `at::empty_like(self, ...).uniform_(0, 1, c10::nullopt)`, the correct way should be `at::empty_like(self).uniform_(0, 1, gen)`. Unfortunately `rand_like` doesn't have generator argument